Simplenote/src/main/java/com/automattic/simplenote/widgets/SimplenoteEditText.java

```diff
@@ -191,28 +191,45 @@ public void run() {
         }
     }
 
-    // Returns the line position of the text cursor
-    private int getCurrentCursorLine() {
-        int selectionStart = getSelectionStart();
-        Layout layout = getLayout();
-
-        if (!(selectionStart == -1)) {
-            return layout.getLineForOffset(selectionStart);
+    /**
+     *  Finds the start of line containing the start of selection
+     */
+    private int findStartOfLineOfSelection() {
+        int position = getSelectionStart();
+        if(position == -1) {
+            position = 0;
+        }
```

Comment:
```suggestion
         // Since the minimum position index is 0 and the selection start can return -1 for no
        // selection, get the maximum value between 0 and getSelectionStart().
        int position = Math.max(0, getSelectionStart());
```

We can use `Math.max` as a trick similar to the one in `findEndOfLineOfSelection()`.  It may not be obvious why we're checking for `-1` if someone isn't familiar with `getSelectionStart()`.  So, we may want to add a comment explaining what's happening.

It's debatable whether using `Math.max` or `if (position == -1)` is better.  Changing this to `Math.max` is not necessary.  It's just a suggestion.  I wanted to point out a possible alternative in case you want to use it.  Either way, we should probably add some comments so that the logic is clear.

Reply:
I adjusted the logic to return 0 directly without looping `if (position == -1)`, IMO it's better, and follows the return early pattern.